3.0/Dockerfile

```diff
@@ -1,66 +1,90 @@
+libboost-filesystem-dev libexpat1-dev zlib1g-dev libxml2-dev \
+libpq-dev libgeos-dev libgeos++-dev libproj-dev libbz2-dev \
+postgresql-server-dev-9.5 git
+# Cleanup is not necessary with multi-stage build!  these dependencies will not be copied over!
 
-RUN sed -i -e"s/^max_connections = 100.*$/max_connections = 1000/" /etc/postgresql/9.5/main/postgresql.conf
+WORKDIR /app/src
 
-# Nominatim install
+# Nominatim install and build
 ENV NOMINATIM_VERSION v3.0.1
-RUN git clone --recursive https://github.com/openstreetmap/Nominatim ./src
-RUN cd ./src && git checkout tags/$NOMINATIM_VERSION && git submodule update --recursive --init && \
-    mkdir build && cd build && cmake .. && make
+RUN git clone --recursive https://github.com/openstreetmap/Nominatim .
+RUN git checkout tags/$NOMINATIM_VERSION && git submodule update --recursive --init
+RUN mkdir build && cd build && cmake .. && make
+
+# RUNTIME!
+FROM ubuntu:xenial
+
+# TODO Do we want to drop the /src at this point?
+WORKDIR /app/src
+COPY --from=0 /app/src .
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV LANG C.UTF-8
+
+RUN apt-get -y update -qq
+RUN apt-get -y install locales
+RUN locale-gen en_US.UTF-8
+RUN update-locale LANG=en_US.UTF-8
 
-# Osmium install to run continuous updates
+# Load initial data
+COPY country_osm_grid.sql.gz data/country_osm_grid.sql.gz
+COPY data.osm.pbf data/data.osm.pbf
+ADD tiger.tar.gz data/
+
+# Runtime Dependencies
+RUN apt-get install -y apache2 php php-pgsql libapache2-mod-php php-pear php-db \
+php-intl python-pip libboost-python-dev gdal-bin python-gdal \
+postgresql-9.5-postgis-2.2 postgresql-contrib-9.5 \
+libbz2-dev libz-dev unzip sudo
+
+# TODO Possibly Unnecessary Dependencies?
+RUN apt-get install -y osmosis
 RUN pip install osmium
 
-# Apache configure
-COPY local.php /app/src/build/settings/local.php
+# Cleanup runtime after dependecy install
+RUN apt-get clean
+RUN rm -rf /var/lib/apt/lists/*
+RUN rm -rf /tmp/* /var/tmp/*
+
+# Apache
+COPY local.php build/settings/local.php
 COPY nominatim.conf /etc/apache2/sites-enabled/000-default.conf
 
-# Load initial data
-ARG PBF_DATA=http://download.geofabrik.de/north-america-latest.osm.pbf
-ARG TIGER_DATA=https://nominatim.org/data/tiger2017-nominatim-preprocessed.tar.gz
-RUN curl -L -f $PBF_DATA --create-dirs -o /app/src/data.osm.pbf
-RUN curl -L -f $TIGER_DATA --create-dirs -o /app/src/data/tiger2017.tar.gz
-RUN curl http://www.nominatim.org/data/country_grid.sql.gz > /app/src/data/country_osm_grid.sql.gz
-RUN tar -xzf /app/src/data/tiger2017.tar.gz -C /app/src/data/
+# Postgres
+RUN echo "host all  all    0.0.0.0/0  trust" >> /etc/postgresql/9.5/main/pg_hba.conf
+RUN echo "listen_addresses='*'" >> /etc/postgresql/9.5/main/postgresql.conf
+RUN sed -i -e"s/^max_connections = 100.*$/max_connections = 1000/" /etc/postgresql/9.5/main/postgresql.conf
+
+# nominatim user
+# TODO Do we need the nominatim user?  This is docker...
```

Comment:
It is possible that this is necessary for Postgres to establish local connections [(search here for "ident")](https://www.davidpashley.com/articles/postgresql-user-administration/).  I wouldn't be surprised if Nominatim (looks like old-school php) is making sql connections to Postgres with a username and password. Not 100% sure on this, haven't examined the code.